kinto/plugins/history/views.py

```diff
@@ -31,3 +32,15 @@ class History(resource.ShareableResource):
     def get_parent_id(self, request):
         self.bucket_id = request.matchdict['bucket_id']
         return instance_uri(request, 'bucket', id=self.bucket_id)
+
+    def _extract_filters(self, queryparams=None):
+        filters = super(History, self)._extract_filters(queryparams)
+
+        for filt in filters:
+            if filt.field in ['record_id', 'collection_id', 'bucket_id']:
+                if isinstance(filt.value, int):
+                    filt_id = Filter(filt.field, str(filt.value), filt.operator)
+                    filters.remove(filt)
+                    filters.insert(0, filt_id)
```

Comment:
Although this is absolutely correct, I would recommend to avoid the *mutation* of the original list, and build a new list instead. It is generally considered as a good practice to avoid *side-effects* on the input variables it receives.

```py
filters = super(...)
filters_str_id = []
for filt in filters:
    if filt.field in (...):
        filt = Filter(..., str(filt.value), ...)
    filters_str_id.append(filt)       
return filters_str_id
```